api/v1beta1/openstackdataplanenodeset_types.go

```diff
@@ -177,14 +185,49 @@ func (instance OpenStackDataPlaneNodeSet) GetAnsibleEESpec() AnsibleEESpec {
 	}
 }
 
-// DataplaneAnsibleImageDefaults default images for dataplane services
-type DataplaneAnsibleImageDefaults struct {
-	Frr                  string
-	IscsiD               string
-	Logrotate            string
-	NeutronMetadataAgent string
-	NovaCompute          string
-	NovaLibvirt          string
-	OvnControllerAgent   string
-	OvnBgpAgent          string
+// DataplaneAnsibleContainerImages default images for dataplane services
+type DataplaneAnsibleContainerImages struct {
+	Frr                  string `json:"frr,omitempty"`
+	IscsiD               string `json:"iscsiD,omitempty"`
+	Logrotate            string `json:"logrotate,omitempty"`
+	NeutronMetadataAgent string `json:"neutronMetadataAgent,omitempty"`
+	NovaCompute          string `json:"novaCompute,omitempty"`
+	NovaLibvirt          string `json:"novaLibvirt,omitempty"`
+	OvnControllerAgent   string `json:"ovnControllerAgent,omitempty"`
+	OvnBgpAgent          string `json:"ovnBgpAgent,omitempty"`
+}
+
+const (
+	// FrrDefaultImage -
+	FrrDefaultImage = "quay.io/podified-antelope-centos9/openstack-frr:current-podified"
+	// IscsiDDefaultImage -
+	IscsiDDefaultImage = "quay.io/podified-antelope-centos9/openstack-iscsid:current-podified"
+	// LogrotateDefaultImage -
+	LogrotateDefaultImage = "quay.io/podified-antelope-centos9/openstack-cron:current-podified"
+	// NeutronMetadataAgentDefaultImage -
+	NeutronMetadataAgentDefaultImage = "quay.io/podified-antelope-centos9/openstack-neutron-metadata-agent-ovn:current-podified"
+	// NovaComputeDefaultImage -
+	NovaComputeDefaultImage = "quay.io/podified-antelope-centos9/openstack-nova-compute:current-podified"
+	// NovaLibvirtDefaultImage -
+	NovaLibvirtDefaultImage = "quay.io/podified-antelope-centos9/openstack-nova-libvirt:current-podified"
+	// OvnControllerAgentDefaultImage -
+	OvnControllerAgentDefaultImage = "quay.io/podified-antelope-centos9/openstack-ovn-controller:current-podified"
+	// OvnBgpAgentDefaultImage -
+	OvnBgpAgentDefaultImage = "quay.io/podified-antelope-centos9/openstack-ovn-bgp-agent:current-podified"
+)
+
+// SetupDefaults - initializes any CRD field defaults based on environment variables (the defaulting mechanism itself is implemented via webhooks)
+func SetupDefaults() {
+	// Acquire environmental defaults and initialize dataplane defaults with them
+	dataplaneAnsibleContainerImagesDefaults = DataplaneAnsibleContainerImages{
```

Comment:
I copied this whole pattern from openstack-ansibleee-operator. Although, in the SetupDefaults there, `:=` is used, where here I have `=`. You're saying that this is the same variable as defined in openstackdataplanenodeset_webhook.go?

I can test if that's the case. I'll check some other operators as well to see how it's handled there.

Reply:
openstack-operator matches the pattern from openstack-ansibleee-operator. So really, the only difference between that pattern and what I did here is that I mistakenly used `=` here instead of `:=`. It actually seems like using `=` would be simpler wouldn't it? Then I could apply the rest of your suggestions.